oshi-core/src/main/java/oshi/software/os/unix/solaris/SolarisOperatingSystem.java

```diff
@@ -129,32 +131,43 @@ public OSProcess getProcess(int pid) {
 
     @Override
     public List<OSProcess> queryDescendantProcesses(int parentPid) {
-        Set<String> descendantPids = new HashSet<>();
-        addChildrenToDescendantSet(Integer.toString(parentPid), descendantPids, true);
+        Set<String> descendantPids = getChildrenOrDescendants(Integer.toString(parentPid), true);
```

Comment:
I missed this in the earlier discussion.  There's no need to convert `parentPid` to a String here just to convert it back in the later method.